main.cc

```diff
@@ -1047,6 +1044,27 @@ To start the scylla server proper, simply invoke as: scylla server (or just scyl
             // since some features affect storage.
             db::system_keyspace::enable_features_on_startup(feature_service).get();
 
+            supervisor::notify("starting schema commit log");
+            auto sch_cl = db.local().schema_commitlog();
+            if (sch_cl != nullptr) {
+                auto paths = sch_cl->get_segments_to_replay();
+                if (!paths.empty()) {
+                    supervisor::notify("replaying schema commit log");
+                    auto rp = db::commitlog_replayer::create_replayer(db).get0();
+                    rp.recover(paths, db::schema_tables::COMMITLOG_FILENAME_PREFIX).get();
+                    supervisor::notify("replaying schema commit log - flushing memtables");
+                    db.invoke_on_all([] (replica::database& db) {
+                        return db.flush_all_memtables();
+                    }).get();
+                    supervisor::notify("replaying schema commit log - removing old commitlog segments");
+                    //FIXME: discarded future
+                    (void)sch_cl->delete_segments(std::move(paths));
```

Comment:
Why not wait?

Reply:
And don't we need to reload the schema?

Reply:
This mirrors the existing logic for regular replay. I'm not sure why it doesn't wait.

We don't need to reload the schema because the schema is loaded only after replay.

Reply:
ok & ok